src/runtime_src/core/pcie/windows/device_windows.cpp

```diff
@@ -911,6 +929,58 @@ struct data_retention
 
 };
 
+struct mailbox
+{
+    using result_type = boost::any;
+
+    static xcl_mailbox
+        init_mailbox_info(const xrt_core::device* dev)
+    {
+        xcl_mailbox info = { 0 };
+        userpf::get_mailbox_info(dev->get_user_handle(), &info);
+        return info;
+    }
+
+    static result_type
+        get_info(const xrt_core::device* device, key_type key)
```

Comment:
Such an uncommented complex code should be just banned by a precommit check right-away.

Reply:
Hi @keryell , thanks for your review. If you go through this device_windows.cpp file, there are no comments before check-ins as well. The one who designed this framework may comment on the complexity of the code.

I am not expert in C++ and I am trying to extend few "xbutil" sub commands as per requests. So, mostly copying the code from the same file and modify it to match my requirement to fix the problem. Similarly, this same answer apply to your other query "why if is not used in place of switch".

If I use if then someone who is a C++ expert can directly suggest me to go through already existing code and try to add similar syntax. So, on the safer side, I mostly follow same syntax of the already existing pieces of code.
Still, I am receiving comments :)